visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -15,119 +15,117 @@ export interface CCValidationResult {
+		message: "API Version is empty or invalid"
+	},
+	majorApiVersionIsOutdated: {
+		title: "Error Major API Version",
+		message: "API Version Outdated: Update CodeCharta API Version to match cc.json"
+	},
+	minorApiVersionOutdated: {
+		title: "Warning Minor API Version",
+		message: "Minor API Version Outdated"
+	},
+	nodesNotUnique: {
+		title: "Error Node Uniques",
+		message: "node names in combination with node types are not unique"
+	},
+	validationError: {
+		title: "Error Validation"
 	}
+}
 
-	public static validate(file: { apiVersion: string; nodes: CodeMapNode[] }): CCValidationResult {
-		const result: CCValidationResult = { error: [], warning: [], title: "" }
-
-		switch (true) {
-			case !file:
-				result.error.push(this.ERROR_MESSAGES.fileIsInvalid.message)
-				result.title = this.ERROR_MESSAGES.fileIsInvalid.title
-				break
-			case !this.isValidApiVersion(file):
-				result.error.push(this.ERROR_MESSAGES.apiVersionIsInvalid.message)
-				result.title = this.ERROR_MESSAGES.apiVersionIsInvalid.title
-				break
-			case this.fileHasHigherMajorVersion(file):
-				result.error.push(this.ERROR_MESSAGES.majorApiVersionIsOutdated.message)
-				result.title = this.ERROR_MESSAGES.majorApiVersionIsOutdated.title
-				break
-			case this.fileHasHigherMinorVersion(file):
-				result.warning.push(this.ERROR_MESSAGES.minorApiVersionOutdated.message)
-				result.title = this.ERROR_MESSAGES.minorApiVersionOutdated.title
-				break
-		}
+export function validate(file: { apiVersion: string; nodes: CodeMapNode[] }): CCValidationResult {
+	const result: CCValidationResult = { error: [], warning: [], title: "" }
+
+	switch (true) {
+		case !file:
+			result.error.push(ERROR_MESSAGES.fileIsInvalid.message)
+			result.title = ERROR_MESSAGES.fileIsInvalid.title
+			break
+		case !isValidApiVersion(file):
+			result.error.push(ERROR_MESSAGES.apiVersionIsInvalid.message)
+			result.title = ERROR_MESSAGES.apiVersionIsInvalid.title
+			break
+		case fileHasHigherMajorVersion(file):
+			result.error.push(ERROR_MESSAGES.majorApiVersionIsOutdated.message)
+			result.title = ERROR_MESSAGES.majorApiVersionIsOutdated.title
+			break
+		case fileHasHigherMinorVersion(file):
+			result.warning.push(ERROR_MESSAGES.minorApiVersionOutdated.message)
+			result.title = ERROR_MESSAGES.minorApiVersionOutdated.title
+			break
+	}
 
-		if (result.error.length === 0) {
-			const ajv = new Ajv({ allErrors: true })
-			const validate = ajv.compile(jsonSchema)
-			const valid = validate(file)
-
-			if (!valid) {
-				result.error = validate.errors.map((error: Ajv.ErrorObject) => this.getValidationMessage(error))
-				result.title = this.ERROR_MESSAGES.validationError.title
-			} else if (!FileValidator.hasUniqueChildren(file.nodes[0])) {
-				result.error.push(this.ERROR_MESSAGES.nodesNotUnique.message)
-				result.title = this.ERROR_MESSAGES.nodesNotUnique.title
-			}
+	if (result.error.length === 0) {
+		const ajv = new Ajv({ allErrors: true })
```

Comment:
Ideally, the validation is separated from the schema generation.

Reply:
What are the benefits of doing so?
in our case we could create a file `SchemaGenerator` which exports a function 

```ts
export function generateSchema() {
	const ajv = new Ajv({ allErrors: true })
	return ajv.validate(jsonSchema)
}
```

Reply:
What I meant is that the schema is currently generated inside of a `validate` function (https://github.com/MaibornWolff/codecharta/pull/860/files#diff-0c4adf7ab75be4463a0b0279057f0fe4R44) and that it's best to keep validation function to validating things only :)

I would split this function into a separate step.